### PR TITLE
Improvements before Qt6

### PR DESCRIPTION
**Pull request by YakoYakoYokuYoku**
Thanks for submitting a pull request! Please provide enough information so that others can review your pull request. Additionally, make sure you've done all of these things:

- [x] I've followed the [contributing guidelines](https://github.com/NatronGitHub/Natron/blob/RB-2.4/CODE_OF_CONDUCT.md) to the best of my understanding
- [x] I've read and understood the [contributing guidelines](https://github.com/NatronGitHub/Natron/blob/RB-2.4/CONTRIBUTING.md)
- [x] I've formatted my code according to Natron's [code style]([#](https://github.com/NatronGitHub/Natron#logistics))
- [x] I've searched the [pull requests tracker](https://github.com/NatronGitHub/Natron/pulls?q=is%3Apr) to ensure that this PR is not a duplicate

## PR Description

**What type of PR is this? (Check one of the boxes below)**

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Improvement (non-breaking change which does not add functionality nor fixes a bug but improves Natron in some way)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] My change requires a change to the documentation
    - [ ] I have updated the documentation accordingly

**What does this pull request do?**

This adds a pair of corrections in the build system needed for landing Qt6 support, they affect mostly the Python bindings generation and are backwards compatible with Qt5.

**Show a few screenshots (if this is a visual change)**

N/A.

**Have you tested your changes (if applicable)? If so, how?**

By building and running Natron.

**Futher details of this pull request**

This is a stopgap before #1019 can be merged.

### CODE REVIEW

**Comment by YakoYakoYokuYoku** [comment]
Will merge tomorrow if no objections are raised.

File: Engine/CMakeLists.txt
Comment (acolwell) [inline Engine/CMakeLists.txt]: How about making a PYSIDE_VERSION or PYSIDE_HEADER_PREFIX var and defining it will all the other stuff in the main CMakefile. That way we can make this non-conditional like all your changes below are?

set(PYENGINE_HEADER PySide${PYSIDE_VERSION}_Engine_Python.h)

File: Gui/CMakeLists.txt
Comment (acolwell) [inline Gui/CMakeLists.txt:33]: Similar comment to the one I made for the Engine CMakefile. Could we make this unconditional here by using a var defined in the top level cmakefile with all the other conditional qt6 stuff.

File: Gui/typesystem_natronGui.xml
Comment (acolwell) [inline Gui/typesystem_natronGui.xml:216]: Is this needed now because this isn't being added by the post processing scripts now?

File: Engine/AppInstance.h
Comment (acolwell) [inline Engine/AppInstance.h:200]: Do we still need to explicitly specify the namespace here and elsewhere NATRON_ENUM was used? I don't believe most(all?) are needed for compiling.

File: tools/utils/runPostShiboken2.sh
Comment (acolwell) [inline tools/utils/runPostShiboken2.sh:59]: It would be nice to preserve this and similar warning disabling/enabling below in the new script. The headers the wrappers generate a TON of warnings that make it really difficult to find the actual issues. Cleaning this up and making it consistent across the different includes would be a nice addition to this PR.

**Review by acolwell — CHANGES_REQUESTED** [review]
In general I'm supportive of these changes and just have a few comments

File: Gui/typesystem_natronGui.xml
Comment (YakoYakoYokuYoku) [reply Gui/typesystem_natronGui.xml:216]: Yes, and that's because during bindings generation the wrapped types are not under a namespace. It's much easier to do it here rather than in post-processing.

File: Engine/AppInstance.h
Comment (YakoYakoYokuYoku) [reply Engine/AppInstance.h:200]: This is needed as is, both namespace and enum specified, or else generation yields wrong results. Even Qt does kinda the same.

File: tools/utils/runPostShiboken2.sh
Comment (YakoYakoYokuYoku) [reply tools/utils/runPostShiboken2.sh:59]: If you don't want to see those warnings then it's best for them to be disabled elsewhere, either in a CMake file or via command line, not in source as it's a kludge and not a proper fix.

Comment (YakoYakoYokuYoku) [reply tools/utils/runPostShiboken2.sh:59]: Perhaps that warning disabling should be added in another PR, out of scope of this one and I don't get those warnings on my machine.

File: Engine/CMakeLists.txt
Comment (YakoYakoYokuYoku) [reply Engine/CMakeLists.txt]: I see your point.

**Comment by devernay** [comment]
@YakoYakoYokuYoku please avoid force-pushing without an approval from the reviewer, except for minor stuff (e.g. typo, or 1-line trivial fix), even if you addressed the reviewer's comments. There's no hurry in merging in general.

**Pull request merged**
